src/overlays/actors/ovl_En_Heishi2/z_en_heishi2.c

```diff
@@ -21,77 +67,880 @@ const ActorInit En_Heishi2_InitVars = {
+            // "Start under the first sleeve!"
+            osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆☆☆☆ １回目袖の下開始！ ☆☆☆☆☆ \n" VT_RST);
+            this->actor.textId = 0x7071; // "Do you really want to go through this gate that much?..."
+            this->unk_30E = 1;
+        } else {
+            // "Start under the second sleeve!"
+            osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆☆☆☆ ２回目袖の下開始！ ☆☆☆☆☆ \n" VT_RST);
+            this->actor.textId = 0x7072; //  "Give me 10 Rupees...Yes...No"
+        }
+        this->unk_300 = 4;
+        this->actionFunc = func_80A5475C;
+
+    }
+
+    else {
+        // "That's okay"
+        osSyncPrintf(VT_FGCOL(CYAN) " ☆☆☆☆☆ それはとおらんよぉ ☆☆☆☆☆ \n" VT_RST);
+        this->unk_300 = 6;
+        this->actor.textId = 0x7029; // "So you say you want to see Princess Zelda, eh?..."
+        this->actionFunc = func_80A5475C;
+    }
+}
+
+void func_80A5344C(EnHeishi2* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 4) && (func_80106BC8(globalCtx) != 0)) {
+
+        this->unk_300 = 5;
+        switch (globalCtx->msgCtx.choiceIndex) {
+            case 0:
+                if (gSaveContext.rupees >= 10) {
+                    Rupees_ChangeBy(-10);
+                    thisx->textId = 0x7098; // "OK, I'll let you pass, quickly!"
+                    this->actionFunc = func_80A53538;
+                    break;
+                }
+
+                else {
+                    thisx->textId = 0x7097; // "Well, maybe that's too much for a kid to pay."
+                    this->actionFunc = func_80A54954;
+                    break;
+                }
+            case 1:
+                thisx->textId = 0x7096; // "Tsk tsk."
+                this->actionFunc = func_80A54954;
+                break;
+
+            default:
+                break;
+        }
+        func_8010B720(globalCtx, thisx->textId);
+    }
+}
+
+void func_80A53538(EnHeishi2* this, GlobalContext* globalCtx) {
+
+    Actor* thisx = &this->actor;
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (this->unk_300 == func_8010BDBC(&globalCtx->msgCtx)) {
+        if (func_80106BC8(globalCtx) != 0) {
+            func_8002DF54(globalCtx, NULL, 8);
+            globalCtx->msgCtx.msgMode = 0x37;
+            this->actionFunc = func_80A535BC;
+        }
+    }
+}
+
+void func_80A535BC(EnHeishi2* this, GlobalContext* globalCtx) {
+    f32 frames = SkelAnime_GetFrameCount(&D_06005500.genericHeader);
+    this->unk_2EC = frames;
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06005500, 1.0f, 0.0f, frames, 2, -10.0f);
+    this->actionFunc = func_80A53638;
+}
+
+void func_80A53638(EnHeishi2* this, GlobalContext* globalCtx) {
+
+    Actor* thisx;
+    f32 frameCount;
+    BgSpot15Saku* gate;
+
+    frameCount = this->skelAnime.animCurrentFrame;
+    thisx = &this->actor;
+    gate = (BgSpot15Saku*)(globalCtx->actorCtx.actorList[7].first);
```

Comment:
youll want to use actor type enum

Reply:
i think this might have been looked over on accident, youll still want to grab the right actor type enum value